### PR TITLE
Initial draft of SuperNova description

### PR DESCRIPTION
**Pull request by adr1anh**

### CODE REVIEW

**Comment by porcuquine** [comment]
This looks good. I'll make another pass with probably some very minor comments. Meanwhile, can you add a link to this in the toplevel README?

**Comment by porcuquine** [comment]
This could maybe be a section for such design notes generally, then a bulleted list of contents and brief description, with this document being the only initial content of `/notes`.

File: notes/supernova.md [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	- $U[\mathsf{pc}_i] = U\_{next}$, $W[\mathsf{pc}_i] = W\_{next}$
+	- $(u',w') = (u'\_{next}, w'\_{next})$ 
+- Save $z\_{i+1},z'\_{i+1}, \mathsf{pc}\_{i+1}$ as inputs for the next iteration. 
+
+
+
+In pseudocode, `prove_step` looks something like:
+
+```
+if i = 0 {
```
Comment (huitseeker) [inline notes/supernova.md:53]: A [Mermaid](https://github.blog/2022-02-14-include-diagrams-markdown-files-mermaid/) diagram to show the flow of the protocol might help, perhaps more than pseudocode?

**Review by huitseeker — COMMENTED** [review]
This looks pretty good to me! Note that with a shorter pseudocode section, and some use of the [doc attribute](https://doc.rust-lang.org/rustdoc/write-documentation/the-doc-attribute.html#the-doc-attribute) this may serve as the module documentation to the Supernova module.
( `#![doc = include_str!("path/to/doc.md")]` .. though I don't remember if rustdoc output support Mathjax)

File: notes/supernova.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
...
+The main improvement of SuperNova over its predecessor, is to allow each iteration to apply one of several functions to the previous output, whereas Nova only supported the iteration of a single function. 
+
+Let $F_0, \ldots, F\_{\ell-1}$ be folding circuits with the same arity $a$. 
+In the context of SuperNova, this means that each $F_j$ takes $a$ inputs from the previous iteration, and returns $a$ outputs. 
+These circuits implement the `circuit_supernova::StepCircuit` trait, where the main differences with the existing `StepCircuit` trait are
+- The circuit $F_j$ provides its `circuit_index` $j$
+- The `synthesize` function upon input $z_i$ returns the next `program_counter` $\mathsf{pc}\_{i+1}$ alongside the output $z\_{i+1}$. It also accepts the (optional) input program counter $\mathsf{pc}_i$, which can be `None` when $\ell=1$. During circuit synthesis, a constraint enforces $\mathsf{pc}_i \equiv j$. In contrast to the paper, the _predicate_ function $\varphi$ is built into the circuit itself. In other words, we have the signature $(\mathsf{pc}\_{i+1}, z\_{i+1}) \gets F\_{j}(\mathsf{pc}\_{i}, z\_{i})$.
+
+The goal is to efficiently prove the following computation:
+```
```
Comment (huitseeker) [inline notes/supernova.md:27]: You'll need an ignore comment here and in the following code blocks

File: notes/supernova.md [hunk 3]
```diff
@@ -0,0 +1,313 @@
...
+Each iteration proves the verification of a proof for $z_i$ and the correctness of $z\_{i+1}$, ensuring the proving of each step remains constant.
+- **Augmentation Circuit**: In the context of the SuperNova protocol, an augmentation circuit refers to a circuit $F'$ composing $F$ with a circuit which partially verifies the validity of the previous output $z_i$ before running $F(z_i)$. 
+- **NIFS Folding Verifier**: A non-interactive folding scheme is a protocol for efficiently updating a proof $\pi_i$ about an iterated function $z\_{i+1} \gets F(z_i)$ into a new proof $\pi\_{i+1}$, through a process referred to as "folding". 
+By splitting the proof into an instance/witness pair $(u,w) = \pi$, the folding verifier describes an algorithm for verifying that the $u$ component was properly updated. 
+
+## SuperNova vs. Nova
+
+The main improvement of SuperNova over its predecessor, is to allow each iteration to apply one of several functions to the previous output, whereas Nova only supported the iteration of a single function.
```
Comment (huitseeker) [inline notes/supernova.md:18]: you can drop "over its predecessor", redundant with the end of the sentence.

**Review by huitseeker — APPROVED** [review]
I like this, and it's in a good enough shape to start iterating on within the repo.

Nit: I would have put the file as `src/supernova/README.md` but that's not very important.

**Pull request merged**